### PR TITLE
Implemented double linked list in Go

### PR DESCRIPTION
**Pull request by Kally95**

### CODE REVIEW

File: data_structures/double_linked_list.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+func (dll *DoubleLinkedList) PrintDLL() {
+	if dll.Length == 0 {
+		return
+	}
+
+	currentNode := dll.Head
+	for currentNode != nil {
+		if currentNode.Next == nil {
```
Comment (abusomani) [inline data_structures/double_linked_list.go:37]: This is not required as currentNode cannot be nil inside the for loop.

File: data_structures/double_linked_list.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+			fmt.Println(currentNode.Value)
+			break
```
Comment (abusomani) [inline data_structures/double_linked_list.go:39]: Not required

File: data_structures/double_linked_list.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	if dll.Length == 0 {
+		return
+	}
+
+	currentNode := dll.Tail
+	for currentNode != nil {
+		if currentNode.Previous == nil {
+			fmt.Println(currentNode.Value)
+			break
+		}
```
Comment (abusomani) [inline data_structures/double_linked_list.go:56]: Same here. not required

**Review by abusomani — CHANGES_REQUESTED** [review]

File: data_structures/double_linked_list.go (hunk 1 above)
Comment (Kally95) [reply data_structures/double_linked_list.go:37]: I only did this as it prints 5 => 4 => 3 for example, as opposed to 5 => 4 => 3 => leaving a trailing =>

Comment (abusomani) [reply data_structures/double_linked_list.go:37]: For that you can do this:

```go
fmt.Print(currentNode.Value)
if currentNode.Next != nil {
    fmt.Print("=>")
}
currentNode = currentNode.Next
```

**Pull request merged**